### DISCUSSION TITLE
提问和一些反馈/建议

### DISCUSSION THREAD

**Discussion by MudMoverGeorge** [GarinZ/link-map#56] [General]
dev您好，首先感谢您开发这个插件，简直是重度标签癌的救星orz（尤其是其他同类插件大多都停更的状态下

鉴于本人是第一次用github不大熟悉，并且有多个问题，感觉放在issues里头不大合适？所以先在这里开个贴罢。

1.提问：可否用与浏览器所登录的帐号不同的帐号购买pro版？比如我浏览器日常使用google帐号A并保持登录，但Link Map pro用google帐号B购买·登录LM并长期使用？

--------------------------------------------------------------

2.反馈/建议：
a. 在对group中的tabs进行管理时(比如挪上或挪下)，操作完成之后group会自动collapse，无论用鼠标拖拽或者快捷键操作都是如此。这个有些不便，不知是否有办法改进？即操作时group不会collapse。（*经测试 在对group之外的临近tabs进行管理操作时，不会collapse）。

b. shortcut页面似乎无法滑动，只能调整窗口大小来显示全部内容。

c. 光标hover在tab时右边出现的三个快捷按钮中，不知可否给×和垃圾桶添加操作确认提示以防误触呢？
  c1.这些hover图标本身无法再度hover来显示文本说明，虽然也比较形象..但是于chrome store下载的当前版本(1.1.2)中的Tips似乎与功能不符？(“2. ➡️️ Hover to operate quick actions，2.1 ✍️ Click edit to rename a node， 2.2 ❌ Click Delete to delete a node ，2.3 ❌ Collapse the parent node to delete all its children”)

d.不知以后是否会加入添加备注notes的功能呢？

(e. 请问语言是跟着浏览器或系统语言走的么？没有看到能选的地方呢..

最后希望多选功能可以早日上线。

谢谢！

**Comment by GarinZ** [comment]
感谢你的反馈~

您提到的问题：
1. Link Map账户可以和浏览器登录账户不同，PRO的购买记录始终和Link Map的登录账户绑定在一起，所以它们虽然都是Google账户但不会互相影响的

您的反馈/建议：
1. group内tabs进行上下移动后group会收起：我自己做了一下测试但是并没有能复现这个问题，请问是否可以为我提供一点额外信息（比如：操作的gif图或者更详细一点的文字描述，使用的浏览器，当前的操作系统）来帮助我更快复现这个问题。
2. shortcut界面滑动问题：这确实是个应该优化的问题，我会在未来的版本改进它
3. 对于删除/关闭按钮增加确认框： 没问题，我会把它做成可配置的
4. 快捷按钮没有hover说明：当前没有增加Hover提示主要是考虑hover的文字会占用空间，时间长了会有点烦人，我猜测用户使用习惯后就不再需要了就暂时没有添加。（"Hover to operate quick actions"的本意是鼠标悬停在Node上会出现快捷按钮组）
5. 备注notes：现在可以通过快捷键和右键菜单添加notes，您提到的备注notes是指添加长文本么？如果可以详细跟我分享一下我会很感谢。
6. 语言选择：Google目前还没有提供允许Chrome插件动态修改语言的机制，它是强制与操作系统语言关联的，所以暂时还不能选择语言。对于这个问题我也跟难受，如果不使用Chrome内置的多语言接口那么商家到Chrome商店中可能又会有问题。但我会持续了解这方面的方案，能切换语言的话我会第一时间支持这个功能。

---

Thank you for your feedback~

You mentioned the following issues:
1. Link Map account can be different from the browser login account, PRO's purchase record is always bound to the Link Map login account, so they are both Google accounts but will not affect each other

Your feedback/suggestions:
1. group tabs will collapse after moving up and down: I have done some testing myself but I was not able to reproduce this problem, can you provide me with some additional information (e.g. gifs of the operation or more detailed text description, browser used, current OS) to help me reproduce this problem faster.
2. the shortcut interface sliding problem: this is really a problem that should be optimized, I will improve it in future versions
3. add a confirmation box for the delete/close button: no problem, I will make it configurable
4. shortcut button without hover instructions: The current Hover tip is not added mainly because the text of the hover will take up space and will be a bit annoying after a long time, I guess the user will not need it after getting used to it, so I did not add it for now. (The original meaning of "Hover to operate quick actions" is that the mouse hovering over the Node will appear as a group of shortcut buttons)
5. notes notes: now you can add notes via shortcuts and right-click menu, you mentioned notes notes is to add long text? If you can share with me in detail I would appreciate it.
6. language selection: Google currently does not provide a mechanism to allow Chrome plug-ins to dynamically modify the language, it is forced to associate with the operating system language, so for the time being can not choose the language. I also have a hard time with this problem, if you do not use Chrome's built-in multi-language interface, then the merchants to the Chrome Store may have problems. But I will continue to understand this aspect of the program, if I can switch the language, I will be the first to support this feature.

**Reply by MudMoverGeorge** [reply]
感谢您的解答！
已购入Pro版支持哈哈。

关于帐号，有有一个新的问题：如果我想edge和chrome同时使用，可以**同时**使用同一个帐号么？

——————————————————
关于之前的反馈/建议：
1.gif如下。全程没有按其他键收起group：
a.鼠标拖拽的场景，可以看到当我把bbb拖入aaa下层之后，之前打开的ttt group自动收起了
![drag and drop by mouse](https://github.com/GarinZ/link-map/assets/136071384/7ca916fe-74af-4f82-a9a8-86d0129b024e)

b.快捷键操作的场景，可以看到我用ctrl+shift+↑把bbb移到aaa上面之后，之前打开的ttt group还是自动收起了
![move up by keyboard shortcuts](https://github.com/GarinZ/link-map/assets/136071384/05982be6-44e1-4158-9b4e-1ca25861457a)

这个测试是在edge Version 113.0.1774.42 (Official build) (64-bit)下进行的，插件版本是edge官方插件商店下载的1.1.2最新版，OS版本为Win10 Home Version	10.0.19045 Build 19045；在同一台电脑同一个OS的Chrome Version 113.0.5672.93 (Official Build) (64-bit)，使用chrome官方插件商店下载的1.1.2最新版，也是一样的情况。


2.了解！

3.感谢！

4.您说的确实有道理。不过不知Tips里的说明文字是否有误或者是我理解错误哈哈， 2.2 ❌ Click Delete to delete a node ，2.3 ❌ Collapse the parent node to delete all its children” 这两个是分别对应×和垃圾桶图标么？怎么感觉×是在浏览器里关闭相应tab、垃圾桶是删掉node同时关闭tab啊😂

5.您说的右键notes就是edite加上那个在名字里显示的短备注吧？我的想法确实是长备注，至于备注呈现的方式感觉可以是悬浮呈现，和/或邮件菜单里多一个类似windows右键菜单里的“属性”页面，并在其中加入更加可以显示·编辑更长和更多备注的空间。
又或许可以在这个“属性”页面里加入更多的内容，比如添加一个“(分类)标记”(在其他软件里好像叫tag或者label)，这样能给分布在不同位置的页面加入共同的标记，搜索相应“标记”的时候就可以同时展现在一处，而不用物理上改变tab所处的位置·窗口了；您现在的tag功能感觉是插入一行便签似的？不知道我理解的对不对哈。但如果想要这种类似微信收藏夹里的表签搜索功能，感觉可能要个专门的搜索页面显示搜索结果了..
edit：啊我发现其实可以用已有的tags功能用作添加备注的..

6.啊这个就没有办法了，不过对我也不是大问题哈哈。


PS：再次感谢您开发这个插件，我试了几个其他的，甚至因为我打开的标签页数量过于修罗场或者他们版本太旧的原因，压根都无法完成加载哈哈，您的不仅可以而且非常快，价格也很便宜。希望可能的话可以一直维护下去~

**Comment by GarinZ** [comment]
不好意思，之前我忘记回复您了。
- Chrome和Edge包括其他浏览器都是可以共用一个账号的，购买了PRO后在登录对应账号的所有浏览器中都可以使用PRO功能
- 自动节点收起的问题：请您更新到刚刚发布的v1.1.4版本，尝试一下是否还存在这类问题，如果还有问题辛苦您再告诉我一下
- 问题4：这里可能我确实提示的不太明确
  - 2.2的意思是点击垃圾桶，代表删除节点
  - 2.3的意思是把节点收起，再点击垃圾桶，可以批量删除当前节点及其所有子节点
  - 这里主要想介绍**收起**和**展开**时候的逻辑会略有不同，**展开**时的操作只会作用于当前节点，**收起**时的操作会作用于当前节点和它的子节点，起到批量操作的效果。
- 问题5：为节点增加可以编辑更长更多的备注的功能也在Link Map的开发计划中，未来一定会加入类似的功能~

非常感谢您喜欢Link Map，如果有任何建议和问题非常欢迎您继续反馈，我会每条都仔细看并进行回复。

---

Sorry, I forgot to reply to you before.
- Chrome and Edge including other browsers can share the same account, after purchasing PRO, you can use the PRO function in all browsers that are logged into the corresponding account.
- Automatic node put away the problem: please update to the just released v1.1.4 version, try to see if there are still such problems, if there are still problems hard to tell me again
- Question 4: I may not be too clear here
  - 2.2 means click the trash can, which means delete the node
  - 2.3 means to put away the node, and then click the trash can to delete the current node and all its children in bulk
  - Here I want to introduce the logic of **put away** and **expand**, the operation of **expand** will only work on the current node, while the operation of **put away** will work on the current node and its children nodes, which will have the effect of batch operation.
- Question 5: The feature of adding longer and more notes for nodes is also in Link Map's development plan, and similar features will definitely be added in the future~.

Thank you very much for choosingLink Map. if you have any suggestions and questions, you are very welcomed to continue to feedback, I will look at each reply carefully and reply.

**Reply by MudMoverGeorge** [reply]
没事没事
自动收起在新版本中已经解决啦！感谢。
问题4是我误解了，我以为2.2和2.3分别对应的是x和垃圾桶图标呢哈哈。

dev加油，感谢您的辛勤工作拯救重度表签癌患者哈哈
